content/troubleshooting/java/application/Java-EffectsOnAppPerformance.md

```diff
@@ -4,18 +4,17 @@ description: "Tips to improve app performance with the Java agent"
 tags: "troubleshoot java agent performance latency startup"
 -->
 
-As you might expect, Contrast's analysis does make your app run a little slower. The good news is it's generally not enough for anyone to complain about, and the results are definitely worth it.
+Contrast's platform has been designed from the ground up to provide maximum value to customers across the SDLC, while minimizing overhead. The following sections discuss some general considerations, as well as considerations specific to Contrast Assess and Contrast Protect.
 
+## General Considerations
 
 ## Startup Time
 
 When you start your server with Contrast, you'll see a few messages that indicate that Contrast is performing its JIT bytecode transformations to add security sensors to the JVM. This startup JIT period will take 8-12 seconds on a fast developer laptop, and maybe 30-40 seconds on outdated, resource-starved QA server hardware.
 
-## Request Processing Time
+## Performance Tuning Hints
 
-It's probably more important to think about how Contrast affects the round-trip time. In typical applications, Contrast adds around 2x round-trip time to a request that contains a lot of business logic with all the bells and whistles turned on. Round trip times for static resources typically don't get measurably worse. In requests where the total round-trip time is dominated by database or Web Service calls, Contrast's effect will be less noticeable.
-
-If better performance is really important to your environment, consider the following options:
+The following considerations apply to our platform as a whole:
```

Comment:
These suggestions don't tell users how to make these changes, or the implications of doing so. Of course users want a 50% performance boost, but turning off data flow rules eliminates a lot of good analysis that differentiates Contrast from other tools. If we're going to propose these items for consideration we should

* Include the implication of making these tuning changes in addition to the expected performance benefits
* Link users to some documentation on how to make this change (such as the Java agent Configuration page)

Reply:
Agreed

Reply:
Guys this has been sitting around for months - meanwhile the language in our docs continues to be horrible. How about this - to start with please immediately delete the sentence that reads

In typical applications, Contrast adds around 2x round-trip time to a request that contains a lot of business logic with all the bells and whistles turned on.

We have got to have better docs.